test.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+export OOC_LIBS=$(dirname `pwd`)
+ARGS=""
+if [ -e rock_arguments.txt ]
+	then
+		ARGS=$(cat rock_arguments.txt)
+fi
+FLAGS=${@:2}
+TESTS=$@
+case "$1" in
+	"")
+		;&
+	"all")
+		TESTS="./test" 
+		;;
+	*)
+		TESTS="./test/$1"
+		;;
+esac
+if [[ -d $TESTS ]]
+then
+	TESTS=$(find $TESTS -name "*Test.ooc")
+else
+	TESTS=$TESTS"Test.ooc"
+fi
+for TEST in $TESTS
+do 
+	OUTPUT=$(rock -q -lpthread --gc=off $ARGS $FLAGS $TEST 2>&1 )
+	if [[ !( $? == 0 ) ]]
+	then
+		echo -e $OUTPUT
+		echo "Compilation failed of test: $TEST"
+		exit 0
+	fi
+	NAME=${TEST##*/}
+	NAME=${NAME%.*}
+	#echo $TEST
+	./$NAME
+	if [[ !( $? == 0 ) ]]
+	then
+		echo "Failed test: $TEST"
+		exit 0
+	fi
+	rm -f $NAME
+done
+exit 1
```

Comment:
Looks like you've confused 0 and 1.

This is the end of the output of the "passing" Travis build for this commit:
> ImageFile .Failed test: ./test/draw/ImageFileTest.ooc
The command "./test.sh all -Dnogpu" exited with 0.
Done. Your build exited with 0.

In Travis, 0 is good, and 1 is bad.